### PR TITLE
Update `Clear-Host` to simply call `[Console]::Clear()` on all platforms and remove `clear` alias on Unix systems

### LINKED ISSUE #8554: 'clear' should leverage tput reset on *nix
# Summary of the new feature/enhancement

As as user, I want the same 'clear' experience in PowerShell on *nix, as I have in Windows. Currently, if I use 'clear' in Windows, this clears the buffer. Currently, in *nix, it appears to create a gap of blank lines and leaves the buffer mostly intact; so, I can scroll up to see the history.

I _believe_ that this is because pwsh is leveraging terminal and when clear is input by the user, it is passed to the terminal. However, the behaviour one would expect should be the equivalent of 'tput reset' in the terminal window - as this would be the same experience seen in PowerShell in Windows.

# Proposed technical implementation details (optional)

Instead of allowing 'clear' to pass to the terminal, translate the command to 'tput reset' and pass that to the terminal.

### PR DESCRIPTION
**Pull request by SteveL-MSFT**
## PR Summary

Based on @PowerShell/powershell-committee decision, standardizing on `[Console]::Clear()` on all platforms and removing `clear` alias on non-Windows.

Fix https://github.com/PowerShell/PowerShell/issues/8554

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Change is not breaking](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` to the beginning of the title and remove the prefix when the PR is ready.
- **User-facing changes**
    - [x] Not Applicable
    - **OR**
    - [ ] User-facing [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed - Issue link:
- **Testing - New and feature**
    - [ ] Not Applicable or can only be tested interactively
    - **OR**
    - [x] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
        - [ ] [Add `[feature]` if the change is significant or affects feature tests](https://github.com/PowerShell/PowerShell/blob/master/docs/testing-guidelines/testing-guidelines.md#requesting-additional-tests-for-a-pr)

### CODE REVIEW

**Review by daxian-dbw — APPROVED** [review]
LGTM

**Comment by iSazonov** [comment]
> removing clear alias on non-Windows

@SteveL-MSFT Please reword the PR description. We are not "removing on non-Windows" but "restoring on Windows".

**Review by iSazonov — APPROVED** [review]

**Comment by mklement0** [comment]
@iSazonov:

We get a unified way to clear both the current screen and the scrollback buffer.

Personally, I virtually always want the scrollback buffer cleared too.

While this is what you get on Windows already - invariably so - you currently cannot get the same behavior in a consistent, PowerShell-native manner on Unix platforms - that's what `-All` would provide.

**Comment by mklement0** [comment]
@iSazonov:

The benefit is an abstract, general one, for consistency and predictability:

Don't make a command accept parameters that it is not designed to accept; in particular, don't make it _quietly ignore_ such parameters.

**Comment by vexx32** [comment]
In general I would think it best practice for any "official" functions be thoroughly implemented, as @mklement0 says. It sets a bad example if we go around including low-quality code in official distributions, and it goes against years of best practices to widely distribute a function that quietly ignores arbitrary parameters.

**Comment by SteveL-MSFT** [comment]
@iSazonov not sure what you mean by "restoring on Windows" since prior to this PR, the `clear` alias is on Windows, Linux, and macOS.  This PR retains `clear` alias on Windows and removes it from Linux and macOS.

@mklement0 personally, I think we can consider extending `Clear-Host` to have a `-IncludeScrollbackBuffer` type of switch, but that changing it to an advanced function is outside the scope of this PR.  I would suggest opening a new issue.  If we go that route, I would suggest having it as an actual C# cmdlet which can be extended.

**Comment by iSazonov** [comment]
@SteveL-MSFT Sorry, I lost the context.

File: src/System.Management.Automation/engine/InitialSessionState.cs [hunk 1]
```diff
@@ -4065,34 +4065,16 @@ internal void ImportCmdletsFromAssembly(Assembly assembly, PSModuleInfo module)
...
-";
-            }
-            else
-            {
-                // Porting note: non-Windows platforms use `clear`
-                return @"
-& (Get-Command -CommandType Application clear | Select-Object -First 1).Definition
+            return @"
+[Console]::Clear()
 # .Link
```
Comment (BrucePay) [inline src/System.Management.Automation/engine/InitialSessionState.cs]: This isn't going to work for non-console hosts like ISE, VSCode(?) and remoting. On those platforms, you still need to use the `$RawUI` portability APIs. A nice solution would be to actually add a new `RawUI.Clear()` API and then implement that using `Console.Clear()` in the console host but that may end up being a bunch of work...

**Pull request merged**